scylla/tests/integration/query_result.rs

```diff
@@ -295,3 +296,171 @@ async fn execute_iter_serialization_error() {
+    session.await_schema_agreement().await.unwrap();
+
+    for i in 0..insert_num {
+        if use_prepared_statements {
+            let prepared_statement = session
+                .prepare(format!("INSERT INTO {}(id, val) VALUES (?, ?)", table_name))
+                .await
+                .unwrap();
+            session
+                .execute_unpaged(&prepared_statement, &vec![i as i32, i as i32])
+                .await
+                .unwrap();
+        } else {
+            let cql = format!("INSERT INTO {}(id, val) VALUES ({}, {})", table_name, i, i);
+            session.query_unpaged(cql, &[]).await.unwrap();
+        }
+    }
+
+    let query_result: QueryResult;
+    if use_prepared_statements {
+        let prepared_statement = session
+            .prepare(format!("SELECT * FROM {}", table_name))
+            .await
+            .unwrap();
+        query_result = session
+            .execute_unpaged(&prepared_statement, &[])
+            .await
+            .unwrap();
+    } else {
+        let select_query = Query::new(format!("SELECT * FROM {}", table_name)).with_page_size(5);
+        query_result = session.query_unpaged(select_query, &[]).await.unwrap();
+    }
+    let rows = query_result.into_rows_result().unwrap();
+
+    // NOTE: check rows number using 'rows_num()' method.
+    assert_eq!(rows.rows_num(), insert_num as usize);
+
+    // NOTE: check actual rows number.
+    let actual_rows = rows
+        .rows::<(i32, i32)>()
+        .unwrap()
+        .collect::<Result<Vec<_>, _>>()
+        .unwrap();
+    assert_eq!(actual_rows.len(), insert_num as usize);
+}
+
+async fn unpaged_error(use_prepared_statements: bool) {
+    let table_name = if use_prepared_statements {
+        "execute_unpaged"
+    } else {
+        "query_unpaged"
+    };
+    let session: Session = create_session(table_name).await;
+    session.await_schema_agreement().await.unwrap();
+
+    let query_result: Result<QueryResult, ExecutionError>;
+    if use_prepared_statements {
+        let prepared_statement = session
+            .prepare(format!("SELECT * FROM {}", table_name))
+            .await
+            .unwrap();
+        // NOTE: drop table to make the main query return error
+        session
+            .ddl(format!("DROP TABLE IF EXISTS {}", table_name))
+            .await
+            .unwrap();
+        query_result = session.execute_unpaged(&prepared_statement, &[]).await;
+    } else {
+        let select_query = Query::new(format!("SELECT * FROM fake{}", table_name));
+        query_result = session.query_unpaged(select_query, &[]).await;
+    }
+    match query_result {
+        Ok(_) => panic!("Unexpected success"),
+        Err(err) => println!("Table not found as expected: {:?}", err),
+    }
+}
+
+#[tokio::test]
+async fn test_query_unpaged_error() {
+    unpaged_error(false).await
+}
+
+#[tokio::test]
+async fn test_execute_unpaged_error() {
+    unpaged_error(true).await
+}
```

Comment:
I'd like to not make tests unnecessarily slow, which means reducing the amount of schema changes.
Those 2 tests could be merged into one which:
- Creates KS and table
- Makes the unprepared check (because it does not remove the table)
- Makes the prepared check.

Reply:
Answered below.